src/frontend/ServerLoading.tsx

```diff
@@ -38,12 +40,8 @@ export const ServerLoading = ({
   }
 
   if (serverStatus.value === 'ERROR') {
-    return (
-      <SafeAreaView
-        style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
-        <Text>Error!</Text>
-      </SafeAreaView>
-    );
+    BootSplash.hide();
+    return <FatalError />;
```

Comment:
We could possibly put this in an error boundary, but I am unsure of the best practices with fatal errors, so im curious what you think @achou11 .

All that being said, even if this was encapsulated in an error boundary, I think we still would need to put this `FatalError` Component here as well. `BootSplash.hide()` is async, so throwing an error after `hide()` would not get caught by the ErrorBoundary, as they are only captured within the react render cycle

Reply:
i think it makes sense here. the error boundary is more for react rendering errors that come up, whereas this is capturing an error happening outside of react (the server). mapeo mobile essentially does what you're doing here, if that's any confirmation 😄